### PR TITLE
MTN add alphacsc recipee 

### PR DESCRIPTION
**Pull request by tomMoral**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

- python `@conda-forge/help-python`
- python/c hybrid `@conda-forge/help-python-c`

### CODE REVIEW

File: recipes/alphacsc/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    - numpy                                  # [build_platform != target_platform]
+    - {{ compiler('c') }}
+  host:
+    - python
+    - pip
+    - numpy
+    - cython
+  run:
+    - python
+    - mne
```
Comment (hoechenberger) [inline recipes/alphacsc/meta.yaml:33]: Would `mne-base` suffice?
```suggestion
    - mne-base
```

File: recipes/alphacsc/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+source:
+  url: https://github.com/alphacsc/alphacsc/archive/{{ version }}.tar.gz
+  sha256: 008acfcd39e8f9b926a22535015ae09a4e6e85986d496d4b279178b396a1dda2
+  patches:
+   - patches/001-remove_sporco_files_from_package.patch
+
+build:
+  number: 0
+  # By default, the package will be built for the Python versions supported by conda-forge and for all major OSs.
+  # Add the line "skip: True  # [py<35]" (for example) to limit to Python 3.5 and newer, or "skip: True  # [not win]" to limit to Windows.
```
Comment (hoechenberger) [inline recipes/alphacsc/meta.yaml:16]:
```suggestion
```

File: recipes/alphacsc/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    - joblib
+    - matplotlib-base
+    - scikit-learn
+
+test:
+  imports:
+    - alphacsc
+  requires:
+    - pip
+    - pytest
```
Comment (hoechenberger) [inline recipes/alphacsc/meta.yaml:46]: You don't really want to run an entire test suite here. Just testing whether some basic imports work is sufficient.

Comment (hoechenberger) [reply recipes/alphacsc/meta.yaml:46]: Oh and the `pip check` of course is always a good idea.

File: recipes/alphacsc/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  host:
+    - python
+    - pip
+    - numpy
+    - cython
+  run:
+    - python
+    - mne-base
+    - numba
+    - numpy
```
Comment (hoechenberger) [inline recipes/alphacsc/meta.yaml:33]: We need to ensure we'll have a compatible NumPy at runtime
```suggestion
    - {{ pin_compatible('numpy') }}
```

File: recipes/alphacsc/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    - python                                 # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+    - cython                                 # [build_platform != target_platform]
+    - numpy                                  # [build_platform != target_platform]
```
Comment (hoechenberger) [inline recipes/alphacsc/meta.yaml:22]: Are you sure all of these are needed here?

Comment (tomMoral) [reply recipes/alphacsc/meta.yaml:22]: I am actually not sure of what is needed.
At the very least, `numpy` and `cython` are. `python` too to run them and the compiler.
So I will keep them for now.

Comment (hoechenberger) [reply recipes/alphacsc/meta.yaml:22]: I'd suggest to try and remove all but the C compiler from this section and see if the build still works

**Comment by carterbox** [comment]
@conda-forge-admin please restart ci

**Review by carterbox — APPROVED** [review]
Merge if builds pass and author doesn't want help with cross-compile.

**Comment by carterbox** [comment]
@tomMoral please mention help-python-c in your new feedstock if you have questions about cross-compiling.

**Pull request merged**